terraform/environments/corporate-staff-rostering/ssm-documents/ami-build.yaml

```diff
@@ -0,0 +1,176 @@
+            Write-Output "Available disk space is less than 50%. Stopping doc execution."
+            Exit 1
+          }
+
+  - name: stopDiscoveryAgent
+    action: aws:runPowerShellScript
+    inputs:
+      runCommand:
+        - |
+          Stop-Service -Name 'AWSDiscoveryService' -ErrorAction SilentlyContinue
+
+  - name: UninstallDiscoveryAgent
+    action: aws:runPowerShellScript
+    inputs:
+      runCommand:
+        - |
+          if (Test-Path 'C:\Program Files\Amazon\Amazon AppStream\aws-discovery-agent\uninstall.exe') {
+            Start-Process 'C:\Program Files\Amazon\Amazon AppStream\aws-discovery-agent\uninstall.exe' -ArgumentList '/S' -Wait
+          } elseif (Test-Path 'C:\Program Files\AWS\AWS Discovery Agent\uninstall.exe') {
+            Start-Process 'C:\Program Files\AWS\AWS Discovery Agent\uninstall.exe' -ArgumentList '/S' -Wait
+          } else {
+            Write-Host "AWS Discovery Agent not found for uninstallation."
+          }
+
+  - name: verifyUninstallation
+    action: aws:runPowerShellScript
+    inputs:
+      runCommand:
+        - |
+          if (-not (Test-Path 'C:\Program Files\Amazon\Amazon AppStream\aws-discovery-agent\uninstall.exe') -and -not (Test-Path 'C:\Program Files\AWS\AWS Discovery Agent\uninstall.exe')) 
+          {
+            Write-Host "AWS Discovery Agent is successfully uninstalled."
+          } else {
+            Write-Host "AWS Discovery Agent uninstallation may have failed."
+          }
+    onFailure: Stop
+
+  - name: InstallFirefoxBrowser
+    action: aws:runPowerShellScript
+    inputs:
+      runCommand:
+        - |
+          Invoke-WebRequest -URI "https://download.mozilla.org/?product=firefox-esr-next-latest-ssl&os=win64&lang=en-GB" -OutFile "$ENV:TEMP\MozillaFirefox.exe"
+          Invoke-Item "$ENV:TEMP\MozillaFirefox.exe"
+
+# wireshark
+
+  - name: InstallNmapTool
+    action: aws:runPowerShellScript
+    inputs:
+      runCommand:
+        - |
+          Invoke-WebRequest -URI "https://nmap.org/download" -OutFile "$ENV:TEMP\nmap.exe"
+          Invoke-Item "$ENV:TEMP\nmap.exe"
+
+  - name: checkADModulePresent
+    action: aws:runPowerShellScript
+    inputs: 
+      runCommand:
+        - |
+          $module = Get-Module -ListAvailable -Name ActiveDirectory
+          if (-not $module) {
+            Write-Error "Active Directory module is not installed. Installing..." 
+          }
+
+  - name: InstallActiveDirectoryModule
+    action: aws:runPowerShellScript
+    inputs:
+      runCommand:
+        - |
+          Install-WindowsFeature -Name "RSAT-AD-PowerShell" -IncludeAllSubFeature
+
+  - name: InstallEC2LaunchV2
+    action: aws:runPowerShellScript
+    inputs: 
+      runCommand:
+        - |
+           Invoke-WebRequest -URI https://s3.amazonaws.com/ec2-downloads-windows/SSM-Agent/EC2Launch/latest/install.ps1 -OutFile $env:TEMP\install.ps1         
+           powershell -ExecutionPolicy Bypass -File $env:TEMP\install.ps1
```

Comment:
It might be better to use `aws:configurePackage` here. You can install a SSM Distributor package directly with that.

https://docs.aws.amazon.com/systems-manager/latest/userguide/documents-command-ssm-plugin-reference.html#aws-configurepackage